core/templates/pages/contributor-dashboard-page/contributor-badge/contributor-badge.component.ts

```diff
@@ -0,0 +1,294 @@
+import { Component, ElementRef, HostListener, ViewChild } from '@angular/core';
+import { AppConstants } from 'app.constants';
+import { LanguageUtilService } from 'domain/utilities/language-util.service';
+import { UserService } from 'services/user.service';
+import { ContributionAndReviewStatsService } from '../services/contribution-and-review-stats.service';
+
+interface ContributionCounts {
+  language?: string;
```

Comment:
Why is language optional?

Reply:
We keep translation and question contribution counts in these objects. Only translation contributions have a language that is associated with it. Hence language is an optional attribute here.

Reply:
Hmm, I think it would be better to use a bit different type then.

Reply:
Or if you want to keep this as optinal then please do 
```suggestion
  language: string | null;
```
And set `null` explicitely.

The other possibility would be to use discriminated unions.

Reply:
Done